CST338/Seperate files/Module4/DataMatrix.java

```diff
@@ -244,22 +288,35 @@ private char readCharFromCol(int col) {
     * to represent our value.
     */
    private boolean writeCharToCol(int col, int code) {
+      // code is asterisk, special column to write
+      if ((col = 0) && (code = 42)) {
```

Comment:
The equal signs here should be double equal signs since comparing equality and not assigning one to the other
if ((col == 0) && (code == 42)) {